### ISSUE TITLE
unable to remove a snapshot with a space in the name

### ISSUE THREAD

**Issue by tchwpkgorg** [lxc/lxd#4431]
It is no longer possible to remove a snapshot with a space in the name.

This used to work in the past, but got broken in the past weeks.

To reproduce:

```# lxc launch images:ubuntu/xenial/amd64 testct

# lxc snapshot testct "2018-04-09 daily"

# lxc info testct
(...)
  2018-04-09 daily (taken at 2018/04/09 15:58 UTC) (stateless)

# lxc delete testvm/2018-04-09\ daily
error: not found

# lxc delete "testvm/2018-04-09 daily"
error: not found

# lxc delete testvm/"2018-04-09 daily"
error: not found
```

```# dpkg -l|grep lxd
ii  lxd                               2.21-0ubuntu3~16.04.1                 amd64        Container hypervisor based on LXC - daemon
ii  lxd-client                        2.21-0ubuntu3~16.04.1                 amd64        Container hypervisor based on LXC - client
```

Running Ubuntu 16.04.

**Comment by brauner** [comment]
This looks related to the db work since mounting and starting the storage volume works just fine. @freeekanayaka

**Comment by brauner** [comment]
```
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/daemon.go: 245: main.(*Daemon).createCmd.func1: handling ip=@ method=GET url=/1.0
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/events.go: 178: main.eventBroadcast.func1: Disconnected event listener: 27516d4e-c94b-427e-9b37-253cf3b912bd
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/daemon.go: 245: main.(*Daemon).createCmd.func1: handling ip=@ method=GET url=/1.0/events
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/events.go: 112: main.eventsSocket: New event listener: 7c42aa05-8a2c-4654-9c98-63d87d41a711
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/daemon.go: 245: main.(*Daemon).createCmd.func1: handling ip=@ method=DELETE url=/1.0/containers/alp1/snapshots/bla+daily
DBUG[04-09|18:32:46] /home/chb/source/go/src/github.com/lxc/lxd/lxd/db/query/retry.go: 22: github.com/lxc/lxd/lxd/db/query.Retry: Database error: &errors.errorString{s:"sql: no rows in result set"}
```

**Comment by brauner** [comment]
Yeah, the url isn't decoded

**Comment by tchwpkgorg** [comment]
Thanks - but how do I use this fix?

If I run lxd 2.21 installed from a deb - do I need to update to lxd from a snap?

**Comment by stgraber** [comment]
We'll cherry-pick this fix into our 3.0 snaps and debs shortly so once we update the backports from 2.21 to 3.0, you'll get the fix.

**Comment by stgraber** [comment]
If it's urgent, I should be able to build you a custom 2.21 binary that you can drop in place.
